jwst/tweakreg/tweakreg_step.py

```diff
@@ -162,6 +166,34 @@ def process(self, input):
                       "containing one or more DataModels.", ) + e.args[1:]
             raise e
 
+        if self.separation <= _SQRT2 * self.tolerance:
+            self.log.error(
+                "Parameter 'separation' must be larger than 'tolerance' by at "
+                "least a factor of sqrt(2) to avoid source confusion."
+            )
+            for model in images:
+                model.meta.cal_step.tweakreg = "SKIPPED"
```

Comment:
Isn't setting "SKIPPED" here is overwritten (again with "SKIPPED") by stpipe since the step attribute "skip" is set and the step defines a "class_alias"?
https://github.com/spacetelescope/stpipe/blob/d1a0a6d25de45400e315afa4fa3cee8633a6b545/src/stpipe/step.py#L484

Reply:
Maybe it does now, I don't think it did when the original code was written. I can remove it or you can do it in your refactoring PR.

Reply:
I'm looking at this after seeing the failure you mentioned https://github.com/spacetelescope/jwst/pull/8476#issuecomment-2116049867

I think I was mistaken and "SKIPPED" is set only when "skip" is step before the step runs.

The changes in this PR would benefit from a unit test that verifies that the parameters that trigger the log message result in the step being skipped (and no error being raised).